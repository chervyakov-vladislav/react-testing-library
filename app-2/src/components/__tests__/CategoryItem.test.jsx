import { screen } from '@testing-library/react';

import { renderWithRouter } from '../../testUtils/routerProvider';
import { CategoryItem } from '../CategoryItem';

describe('CategoryItem', () => { 
  it('рендер компонента', () => {
    renderWithRouter(<CategoryItem strCategory="Cheese" strCategoryThumb="123" strCategoryDescription="/meal.png"/>);

    expect(screen.getByRole('article')).toMatchSnapshot();
  });
});