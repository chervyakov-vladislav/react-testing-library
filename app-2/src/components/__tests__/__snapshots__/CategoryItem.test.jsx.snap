// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem рендер компонента 1`] = `
<div
  class="card"
  role="article"
>
  <div
    class="card-image"
  >
    <img
      alt="Cheese"
      src="123"
    />
  </div>
  <div
    class="card-content"
  >
    <span
      class="card-title"
    >
      Cheese
    </span>
    <p>
      /meal.png
      ...
    </p>
  </div>
  <div
    class="card-action"
  >
    <a
      class="btn"
      href="/category/Cheese"
    >
      Watch category
    </a>
  </div>
</div>
`;
